crates/xlayer/db/src/internal_transaction_inspector.rs

```diff
@@ -0,0 +1,307 @@
+            reth_revm::interpreter::CallScheme::DelegateCall => "delegatecall".to_string(),
+            reth_revm::interpreter::CallScheme::StaticCall => "staticcall".to_string(),
+        };
+
+        let call_input = match &inputs.input {
+            CallInput::SharedBuffer(range) => ctx
+                .local()
+                .shared_memory_buffer_slice(range.clone())
+                .map(|s| Bytes::from(s.to_vec()))
+                .unwrap_or_default(),
+            CallInput::Bytes(b) => b.clone(),
+        };
+
+        self.init_op(
+            call_type,
+            inputs.caller.to_string(),
+            inputs.target_address.to_string(),
+            call_input,
+            inputs.value.get().to_string(),
+            inputs.gas_limit,
+            inputs.bytecode_address.to_string(),
+        );
+
+        self.before_op();
+
+        None
+    }
+
+    fn call_end(&mut self, _ctx: &mut CTX, _inputs: &CallInputs, outcome: &mut CallOutcome) {
+        let trace_index = self.trace_stack.pop().unwrap_or_default();
+        let (_before, after) = self.traces.split_at_mut(trace_index);
+        let (txn, remainder) = after.split_first_mut().unwrap();
+        txn.gas_used = outcome.result.gas.spent();
+        txn.output = outcome.result.output.clone();
+        txn.is_error = !outcome.result.is_ok();
+        txn.error =
+            if txn.is_error { Self::format_error(&outcome.result.result) } else { String::new() };
+        if txn.is_error {
+            for within in remainder {
+                within.is_error = txn.is_error;
+                within.error = txn.error.clone();
+            }
+        }
+
+        self.after_op();
+    }
+
+    fn create(&mut self, _ctx: &mut CTX, inputs: &mut CreateInputs) -> Option<CreateOutcome> {
+        let call_type = match inputs.scheme {
+            reth_revm::interpreter::CreateScheme::Create => "create".to_string(),
+            reth_revm::interpreter::CreateScheme::Create2 { salt: _ } => "create2".to_string(), /* code_address */
+            reth_revm::interpreter::CreateScheme::Custom { address: _ } => "custom".to_string(),
+        };
+
+        self.init_op(
+            call_type,
+            inputs.caller.to_string(),
+            "".to_string(),
+            inputs.init_code.clone(),
+            inputs.value.to_string(),
+            inputs.gas_limit,
+            "".to_string(),
+        );
+
+        self.before_op();
+
+        None
+    }
+
+    fn create_end(&mut self, _ctx: &mut CTX, _inputs: &CreateInputs, outcome: &mut CreateOutcome) {
+        let trace_index = self.trace_stack.pop().unwrap_or_default();
+        let (_before, after) = self.traces.split_at_mut(trace_index);
+        let (txn, remainder) = after.split_first_mut().unwrap();
+        txn.to = outcome.address.unwrap_or_default().to_string();
+        txn.gas_used = outcome.result.gas.spent();
+        txn.output = outcome.result.output.clone();
+        txn.is_error = !outcome.result.is_ok();
```

Comment:
```suggestion
        txn.is_error = outcome.result.is_error();
```
I believe we can do this.